app/controllers/concerns/foreman/controller/normalize_vmware_storage_controller_attributes.rb

```diff
@@ -0,0 +1,17 @@
+module Foreman::Controller::NormalizeVmwareStorageControllerAttributes
+  extend ActiveSupport::Concern
+
+  private
+
+  def normalize_vmware_storage_controller_attributes(attrs)
+    ctrls_and_vol = JSON.parse(attrs["controllers"]).
+      deep_transform_keys { |key| key.to_s.underscore }.
+      deep_symbolize_keys
+    volumes = ctrls_and_vol[:volumes].each_with_index.to_h { |vol, index| [index.to_s, vol] }
```

Comment:
Perhaps it makes sense to combine line 13 & this one:
```suggestion
    attrs["volumes_attributes"] = ctrls_and_vol[:volumes].each_with_index.to_h { |vol, index| [index.to_s, vol] }
```